python/ray/data/_internal/datasource/parquet_datasink.py

```diff
@@ -90,14 +119,13 @@ def write_blocks_to_path():
             else:
                 output_schema = user_schema
 
-            if not self.partition_cols:
-                self._write_single_file(
-                    self.path, tables, filename, output_schema, write_kwargs
-                )
-            else:  # partition writes
-                self._write_partition_files(
-                    tables, filename, output_schema, write_kwargs
-                )
+            self._write_parquet_files(
+                tables,
+                filename,
+                output_schema,
+                ctx.kwargs[WRITE_UUID_KWARG_NAME],
```

Comment:
(No need to do in this PR) We should fix this and not pass anything critical like that